lib/services/android/gradle-build-args-service.ts

```diff
@@ -57,15 +56,25 @@ export class GradleBuildArgsService implements IGradleBuildArgsService {
 		this.$projectData.initializeProjectData(buildData.projectDir);
 
 		args.push(
+			`--stacktrace`,
 			`-PcompileSdk=android-${toolsInfo.compileSdkVersion}`,
 			`-PtargetSdk=${toolsInfo.targetSdkVersion}`,
 			`-PbuildToolsVersion=${toolsInfo.buildToolsVersion}`,
 			`-PgenerateTypings=${toolsInfo.generateTypings}`,
+			`-PprojectRoot=${this.$projectData.projectDir}`,
+			`-DprojectRoot=${this.$projectData.projectDir}`, // we need it as a -D to be able to read it from settings.gradle
+			`-PappPath=${this.$projectData.getAppDirectoryPath()}`,
+			`-PappBuildPath=${this.$projectData.getBuildRelativeDirectoryPath()}`,
+			`-DappBuildPath=${this.$projectData.getBuildRelativeDirectoryPath()}`, // we need it as a -D to be able to read it from settings.gradle
 			`-PappPath=${this.$projectData.getAppDirectoryPath()}`,
 			`-PappResourcesPath=${this.$projectData.getAppResourcesDirectoryPath()}`
 		);
 		if (buildData.gradleArgs) {
-			args.push(buildData.gradleArgs);
+			const additionalArgs: string[] = []
+			buildData.gradleArgs.forEach(arg=>{
+				additionalArgs.push(...arg.split(' -P').map((a,i) => i === 0 ? a : `-P${a}`));
```

Comment:
What does this do?

Reply:
If i remember correctly the reason is we need `["-Pa","-P"]` from `"-Pa -Pb"`.
We could also split with `-P` and trim. Maybe it would be easier to read?

Reply:
Yeah I think split on `-P` and trim/map `-P${arg}` would be easier to read (and essentially do the same thing without that `index === 0` check).